### ISSUE TITLE
Question regarding Scaling Laws on PILE

### ISSUE THREAD

**Issue by LeoXinhaoLee** [state-spaces/mamba#179]
Thank you so much for releasing codes for this inspiring work!

I found the scaling laws experiments on The PILE particularly attractive, showing the superiority of Mamba over many modern models. 

(1) I am wondering if the perplexity numbers in Figure 4 are evaluated on the **validation set of PILE, or the training subset of that run**? 

(2) It would also be tremendously helpful if the **specific perplexity numbers** in Figure 4 for all models could be released so that the community could better study and extend Mamba and its experiments.

Thank you very much for your time and help! @tridao

**Comment by albertfgu** [comment]
1. The perplexity numbers are on the validation set.
2. I can try to collate these for the next version of the paper.

**Comment by LeoXinhaoLee** [comment]
A small follow-up question for scaling laws config, regarding "linear learning rate warmup with cosine decay":

<img width="524" alt="image" src="https://github.com/state-spaces/mamba/assets/62500792/7e38d31a-a23e-4ae7-9f47-3ac516bc0264">

(1) cosine decay to lr=0 in the end?

(2) how many steps for warmup?

(3) is weight decay and gradient clipping applied to all kinds of parameters?

(4) do word embedding and classification head share parameters (like in nano-gpt)?

Thank you very much for your time and help!

**Comment by Niccolo-Ajroldi** [comment]
@LeoXinhaoLee This might help: https://github.com/state-spaces/mamba/issues/184
